src/utils/request/fetch.ts

```diff
@@ -229,8 +253,11 @@ export default function fetchRequest(
       throw cancellation;
     }
     deregisterCancelLstnr();
-    if (timeouted) {
-      log.warn("Fetch: Request timeouted.");
+    if (isTimedOut) {
+      log.warn("Fetch: Request timed out.");
+      throw new RequestError(options.url, 0, NetworkErrorTypes.TIMEOUT);
+    } else if (isConnectionTimedOut) {
+      log.warn("Fetch: Request connection timed out.");
       throw new RequestError(options.url, 0, NetworkErrorTypes.TIMEOUT);
```

Comment:
So there's no mean to find on the error whether the timeout is linked to connection establishment or for the whole request.

I don't know if it's something an application could want, but we have to think explicitly about this.

Yet adding a `NetworkErrorType` for a new kind of timeout, break the API for the application each time we do that, so maybe it shouldn't be done through it anyway.

Reply:
I initially have created a type `NetworkErrorType.CONNECTION_TIMEOUT`  but it break the retry (and maybe more?) as this `NetworkErrorType` was not considered in the `shouldRetry` function.

I removed it for the sake of simplicity but I think we can add it back has it give more details on why the request was aborted

Reply:
Yeah but now I'm afraid that it could be seen as an API breakage as what would before just be a `TIMEOUT` could now be a `CONNECTION_TIMEOUT` instead.

In the future we could even have things like `STALL_TIMEOUT`, so this situation may repeat.
Thus I'm not sure using the `NetworkErrorType` there would be best.

Reply:
I reverted the changes